src/stomach.h

```diff
@@ -13,73 +13,70 @@ class JsonIn;
 class JsonOut;
 class item;
 
-// how much the stomach_contents passes
-// based on 30 minute increments
-struct stomach_pass_rates {
-    units::volume min_vol;
-    float percent_vol;
-    int min_kcal;
-    float percent_kcal;
-    int min_vit;
-    float percent_vit;
+// Contains all information that can pass out of (or into) a stomach
+struct nutrients {
+    units::volume water;
+    units::volume solids;
+    int kcal;
+    std::map<vitamin_id, int> vitamins;
 };
 
-// how much a stomach_contents can absorb
+// how much a stomach_contents can digest
 // based on 30 minute increments
-struct stomach_absorb_rates {
+struct stomach_digest_rates {
+    units::volume solids;
+    units::volume water;
     float percent_kcal;
     int min_kcal;
-    std::map<vitamin_id, float> percent_vitamin;
-    std::map<vitamin_id, int> min_vitamin;
-    float percent_vitamin_default;
-    int min_vitamin_default;
+    float percent_vitamin;
+    int min_vitamin;
 };
 
 // an abstract of food that has been eaten.
 class stomach_contents
 {
     public:
         stomach_contents();
-        stomach_contents( units::volume max_volume );
-
-        // empties the stomach_contents of all of its contents
-        void bowel_movement();
-        // empties contents equal to amount as ratio
-        // amount ranges from 0 to 1
-        void bowel_movement( const stomach_pass_rates &rates );
-        // moves @rates contents to other stomach_contents
-        // amount ranges from 0 to 1
-        void bowel_movement( const stomach_pass_rates &rates, stomach_contents &move_to );
+        /**
+         * @brief Construct a new stomach contents object
+         * Stomachs always process their food in a few hours. Guts take significantly longer,
+         * and their rate is dependent on the metabolic rate of their owner.
+         * @param max_volume Base capacity, subject to modification, i.e. by traits
+         * @param is_stomach If true, treated as stomach, if false, treated as guts
+         */
+        stomach_contents( units::volume max_volume, bool is_stomach );
 
         // turns an item into stomach contents
         // will still add contents if past maximum volume.
         void ingest( player &p, item &food, int charges );
 
+        // Directly adds nutrients to stomach contents
+        void ingest( nutrients ingested );
```

Comment:
```suggestion
        void ingest( const nutrients &ingested );
```
no reason not to have const &, it'll be much better to pass a ref than to copy an object